workspaces/global-header/plugins/global-header/src/components/HeaderIcon/HeaderIcon.tsx

```diff
@@ -13,16 +13,17 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import type { CSSProperties } from 'react';
+import type { CSSProperties, ComponentType } from 'react';
 import { useApp } from '@backstage/core-plugin-api';
 import MuiIcon from '@mui/material/Icon';
 import Box from '@mui/material/Box';
+import type { SvgIconProps } from '@mui/material/SvgIcon';
 
 /**
  * @public
  */
 export interface HeaderIconProps {
-  icon: string;
+  icon: string | ComponentType<SvgIconProps>;
```

Comment:
how will/can this configuration work in a yaml file? can I provide custom icons as a customer editing a ConfigMap?

Reply:
is it possible to duplicate the RHDH CommonIcons.tsx instead of needing a separate API for dev-mode icons? i feel like it would be less complex to keep some files in sync vs. maintain two ways to render icons...